prow/jobs/api-gateway/api-gateway-release.yaml

```diff
@@ -0,0 +1,100 @@
+# Code generated by rendertemplates. DO NOT EDIT.
+
+
+postsubmits: # runs on main
+  kyma-project/api-gateway:
+    - name: release-api-gateway
+      annotations:
+        pipeline.trigger: "pr-merge"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "release-api-gateway"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-sa-gcr-push: "true"
+      optional: false
+      skip_report: false
+      decorate: true
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^\d+\.\d+\.\d+(?:-.*)?$
+      spec:
+        containers:
+          - image: "eu.gcr.io/sap-kyma-neighbors-dev/kaniko-build:v20220926-52bb5a7f0"
+            command:
+              - "/kaniko-build"
+            args:
+              - "--name=api-gateway"
+              - "--config=/config/kaniko-build-config.yaml"
+              - "--context=."
+              - "--dockerfile=Dockerfile"
+              - "--tag=${PULL_BASE_REF}"
```

Comment:
I don't think this will work. Since you use `kaniko-build` as an entrypoint, any bash-like structure will simply be treated as string.

If you want this variable substitution to be working, then you have to use `/bin/sh` as command and add to the start of args: `-c` `/kaniko-build`